MovieCave - MVVM/MovieCave/MovieCave/Coordinators/TabBarCoordinator.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  MainCoordinator.swift
+//  MovieCave
+//
+//  Created by Admin on 20.09.23.
+//
+
+import UIKit
+
+protocol TabBarCoordinatorDelegate: NSObject {
+    
+    /// Handles the click event of a tab bar item with the specified item name.
+    /// - Parameter itemName: The name of the tab bar item that was clicked.
+    func tabBarItemClicked(itemName: String)
+}
+
+class TabBarCoordinator: Coordinator, TabBarCoordinatorDelegate {
+
+    //MARK: - Properties
+    private var rootNavController: UINavigationController
+    private let pages: [TabBarPage] = [.movies, .tvSeries, .profile]
+        .sorted(by: {
+            $0.pageOrderNumber() < $1.pageOrderNumber()
+        })
+    
+    //MARK: - Initializer
+    init(rootNavController: UINavigationController) {
+        self.rootNavController = rootNavController
+    }
+    
+    //MARK: - Methods
+    override func start() {
+        let controllers: [UINavigationController] = pages.map({ getTabController($0) })
+
+        prepareTabBarController(withTabControllers: controllers)
+        identifier = Constants.tabBarCoordinatorID
+        parentCoordinator?.childCoordinators.removeAll {
```

Comment:
I don't like that the removal of the login coordinator is happening here. Can't you do it elsewhere in a place it makes more sense?